File: C/insercion_secuencial.c
// Programa para buscar un elemento de forma secuencial y ordenarlo por insertion sort

#include <stdio.h>

// Funcion para llenar un arreglo

void fill_array(int arr[], int size)
{
    register int i;
    int element;

    for (i = 0; i < size; i++)
    {
        printf("\nIngrese el valor del elemento %d ", i);
        scanf("%d", &element);
        arr[i] = element;
    }
    printf("\n");
}

// Funcion para imprimir un arreglo

void print_array(int arr[], int size)
{
    register int i;

    printf("\nLos elementos del arreglo son: [ ");
    
    for (i = 0; i < size; i++)
    {
        printf("%d ",arr[i]);
    }
    
    printf("]\n");
}

// Funcion para buscar un elemento de forma secuencial

int linear_search(int arr[], int size, int v)
{
    register int i;

    for (i = 0; i < size; i++)
    {
        if (arr[i] == v)
            return i;
    }

    return -1;
}

// Funcion para ordenar el arreglo por insertion sort

void insertion_sort(int arr[], int size)
{
    register int i, j;
    int t;

    for (i = 1; i < size; i++)
    {
        t = arr[i];
        for (j = i; j > 0 && t < arr[j - 1]; j--)
            arr[j] = arr[j - 1];
        arr[j] = t;
    }
}

void main()
{
    int n, pos, value;

    printf("\nIngrese la cantidad de elementos del arreglo: ");
    scanf("%d", &n);

    int array[n];

    // Llamada a funcion para rellenar el arreglo

    fill_array(array, n);
    print_array(array, n);

    // Llamada a funcion para buscar un elemento de forma secuencial

    printf("\nIngrese el valor del elemento a buscar: ");
    scanf("%d", &value);

    pos = linear_search(array, n, value);
    printf("\nLa posicion del elemento es: %d", pos);

    // Llamada a funcion para ordenar el arreglo por insertion sort

    printf("\nOrdenando por insertion sort...\n");
    insertion_sort(array, n);
    print_array(array, n);
}

File: Python/insercion_secuencial.py
# Programa para buscar un elemento de forma secuencial y ordenar la lista mediante insertion sort

# Funcion para llenar una lista

def fill_list(lista):
    length = int(input("\nIngrese la cantidad de elementos de la lista: "))

    for i in range(0, length):
        value = int(input(f"Ingrese el elemento numero {i}: "))
        lista.append(value)

# Funcion para la linear search

def linear_search(lista, size, v):

    for i in range(0, size):
        if lista[i] == v:
            return i
    
    return -1

# Funcion para el insertion sort

def insertion_sort(lista, size):

    for i in range(1, size):
        t = lista[i]
        j = i
        while (j > 0) and (t < lista[j - 1]):
            lista[j] = lista[j - 1]
            j -= 1
        
        lista[j] = t



# Funcion main()

int_list = []

# Llamada a funcion para llenar la lista

fill_list(int_list)
print(f"\nLa lista es: {int_list}")
list_size = len(int_list)

# Llamada a funcion para la linear search

value = int(input("Ingrese el valor del elemento a buscar: "))
pos = linear_search(int_list, list_size, value)
print(f"\nLa posicion del elemento {value} es: {pos}")

# Llamada a funcion para el insertion sort

print("\nOrdenando por insertion sort...\n")
insertion_sort(int_list, list_size)
print(f"La lista es: {int_list}\n")

